pegasus/sites.v3/code.org/public/advocacy/state-facts/splat.haml

```diff
@@ -16,237 +17,297 @@ theme: pdf
+          - if state_info[:percent_ap_female_i] != 0
+            only
+            = succeed "%" do
+              = state_info[:percent_ap_female_i]
+          - else 
+            none
+          were female;
+          - if state_info[:ap_hispanic_i] != 0
+            only
+            = format_integer_with_commas(state_info[:ap_hispanic_i])
+          - else 
+            none
+          were Hispanic;
+          - if state_info[:ap_black_i] != 0
+            only
+            = format_integer_with_commas(state_info[:ap_black_i])
+          - else 
+            none
+          were black.
+      %li
+        - if state_info[:schools_ap_cs_i] != 0
+          Only
+          = format_integer_with_commas(state_info[:schools_ap_cs_i])
+          - if state_info[:schools_ap_cs_i] == 1
+            school
+          - else
+            schools
+          in 
+          = state_info[:state_code_s]
+          = surround "(", "% of" do
+            = state_info[:percent_schools_ap_cs_i]
+          = state_info[:state_code_s]
+          schools with AP programs) offered the AP Computer Science course in 2013-2014.
+        - else
+          No schools in 
+          = state_info[:state_code_s]
+          offered the AP Computer Science course in 2013-2014.
+        - if state_param != 'ND'
+          There are fewer AP exams taken in computer science than in any other STEM subject area.
+
+  %div
+    %h2 What can you do to improve K-12 CS education? 
+    %ol
+      %li
+        %a{href: "https://code.org/promote/letter"}
+          Call
+        on your school to expand computer science offerings at every grade level.
+      %li Ask your local school district to allow computer science courses to satisfy a core math or science requirement.
+      %li
+        Visit
+        %a{href: state_info[:website_url_s]}= state_info[:website_url_s]
+        to learn more about supporting computer science in your state.
+
+  %div
+    %h2
+      Code.org Impact in
       = state_info[:state_name_s]
-      have already trained
-      = state_info[:num_elementary_trained_i]
-      elementary teachers.
-    %li
-      Code.org has trained
-      = state_info[:num_ms_trained_i]
-      middle school teachers and
-      = state_info[:num_hs_trained_i]
-      high school teachers in
+    %ul
+      %li
+        Code.org has partnered with
+        = succeed '.' do
+          = state_info[:school_district_partners_t]
+      %li
+        - if state_info[:num_affiliates_i] == 0
+          There are no Code.org affiliates in the state to offer professional development for elementary teachers.
+        - else
+          - if state_info[:num_elementary_trained_i] == 0
+            [num_affiliates_i] Code.org affiliates offers professional development for elementary teachers in [State name]
+          - else
+            = format_integer_with_commas(state_info[:num_affiliates_i])
+            Code.org
+            - if state_info[:num_affiliates_i] == 1
```

Comment:
i wonder if there's a helper function we could put into an .rb file in which you provide the number and then both strings, and it picks the right one?  (i bet standard loc libraries have something very similar.)